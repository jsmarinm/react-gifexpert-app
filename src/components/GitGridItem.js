import React from 'react'

import PropTypes from 'prop-types';

export const GitGridItem = ({id, title, url}) => {

console.log({id, title , url });

  return (
    <div className="card animate__animated animate__fadeIn">
       <img src={ url } alt={title} />
        <p>{title}</p>
    </div>
  )
}

GitGridItem.protoType = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};