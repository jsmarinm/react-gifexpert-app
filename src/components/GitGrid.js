import React from 'react'
import { useFetchGifts } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

import { GitGridItem } from './GitGridItem';

export const GitGrid = ({category}) => {

    const { data:images, loading } = useFetchGifts(category);

  return (
    <>
        <h3 className="animate__animated animate__fadeIn">{ category }</h3>
        { loading && <p className="animate__animated animate__flash">Loading...</p> }
        <div className="card-grid">
              
            { 
                images.map(img => (
                    <GitGridItem 
                    key = {img.id} 
                    {...img} />
                ))
               
            }
        </div>    
    </>
  )
};

GitGrid.propTypes = {
    category: PropTypes.string.isRequired
}
