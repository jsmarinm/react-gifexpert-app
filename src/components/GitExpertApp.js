//Functional component

import { useState } from "react";
import { AddCategory } from "./AddCategory";
import { GitGrid } from "./GitGrid";

const GitExpertApp = ({ defaultCategories = [] }) => {

    // const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];

    const [categories, setCategories] = useState(defaultCategories);

    // const handleAdd = () => {

    //     // setCategories ([...categories, 'Splider Man']);
    //     setCategories ( cats => [...cats, 'Splider Man']);
    // };

    return(
        <>
            <h2>GitExpertApp</h2>
            <AddCategory setCategories= {setCategories} />
            <hr />
            {/* <button onClick={handleAdd} >Agregar </button> */}

            <ol>
                {
                    // categories.map(category => {
                    //     return <li key={ category }>{ category }</li>
                    // })
                    categories.map(category => (
                    <GitGrid 
                    key={category}
                    category={category} />
                    ))
                    
                 }
               
            </ol>
         
        </>
    );

};


export default GitExpertApp;