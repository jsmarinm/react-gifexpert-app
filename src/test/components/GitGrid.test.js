import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GitGrid } from '../../components/GitGrid';
import { useFetchGifts } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Prueba con GitGrid', () => {

    
  

    test('Debe mostrar correctamente el componente', () =>{
        useFetchGifts.mockReturnValue({
            data: [],
            loading: true
        
        });
        const wrap = shallow(<GitGrid category='One punch man' />);
        expect(wrap).toMatchSnapshot();

    });

    test('debe demostrar items cuando se cargan imagenes useFetchGifts', () => { 

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquiercosa/cosa.jpg',
            title: 'Cualquier cosa'

        },
        {   
        id: 'ABC',
        url: 'https://localhost/cualquiercosa/cosa.jpg',
        title: 'Cualquier cosa'

        }
    ];
        
        useFetchGifts.mockReturnValue({
            data: gifs,
            loading: false
        
        });
        const wrap = shallow(<GitGrid category='One punch man' />);
        // expect(wrap).toMatchSnapshot();
        expect(wrap.find('p').exists()).toBe(false);
        expect(wrap.find('GitGridItem').length).toBe(gifs.length);

     });

});