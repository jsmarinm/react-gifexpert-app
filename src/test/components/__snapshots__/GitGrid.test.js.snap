// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba con GitGrid Debe mostrar correctamente el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One punch man
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
